great_expectations/checkpoint/actions.py

```diff
@@ -251,12 +255,18 @@ def _substitute_config_str_if_needed(value: Union[str, ConfigStr, None]) -> Opti
             return value
 
 
-def _should_notify(success: bool, notify_on: Literal["all", "failure", "success"]) -> bool:
-    return (
-        notify_on == "all"
-        or (notify_on == "success" and success)
-        or (notify_on == "failure" and not success)
-    )
+def should_notify(
+    success: bool, notify_on: NotifyOn, max_severity: Optional[FailureSeverity] = None
+) -> bool:
+    if notify_on in {"all", "success", "failure"}:
+        return (
+            notify_on == "all"
+            or (notify_on == "success" and success)
+            or (notify_on == "failure" and not success)
+        )
+    if success is False and max_severity:
+        return notify_on == max_severity
```

Comment:
I'm guessing this satisfies the ticket's AC, but is this really what we want? This looks like an exact match for worst severity, but wouldn't it make more sense if max_severity were equal to `notify_on` or more severe. I'm guessing this has already been discussed?

Reply:
Sweet, thanks!